### PR TITLE
Tiled gallery block: add babel-polyfill

### LINKED ISSUE #31224: Tiled Gallery Block: Galleries not displaying properly in IE11 on simple sites
It appears the Tiled Gallery block is broken on WPCOM simple sites when viewed in IE11. The images are displayed full-sized without organization.   I've reproduced it on multiple Simple WPCOM sites across multiple themes.

Cannot reproduce on AT or self-hosted.

#### Steps to reproduce
1. Add a Tiled Gallery block to a post/page on a WPCOM simple site.
2. Publish and view page/post directly on IE11
3. Review that the gallery's images are full-sized and not displaying correctly.

#### What I expected
A tiled gallery to be displayed properly as seen here: https://cld.wthms.co/G3axFC

#### What happened instead
Images are displayed full-sized with no distinct organization. https://cld.wthms.co/GjutIH

#### Browser / OS version
IE11 / Windows 10 & IE11 simulators.

#### Screenshot / Video
Expected: https://cld.wthms.co/G3axFC
Result: https://cld.wthms.co/GjutIH

#### Context / Source

#user-report 1851937-zen

### PR DESCRIPTION
**Pull request by simison**
#### Changes proposed in this Pull Request

* Add [Babel polyfill](https://babeljs.io/docs/en/babel-polyfill) to view -side script

#### Testing instructions

- Spin up jurassic ninja with this branch gutenpack-jn
- Add Tiled gallery to a page where no other blocks are not present (to ensure you're not loading `wp-polyfill` to cover the issue, 
- open the saved page in IE11 and observe no more  fatal errors in debugger

Fixes #31224

### CODE REVIEW

File: client/gutenberg/extensions/tiled-gallery/editor.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import '@babel/polyfill';
```
Comment (simison) [inline client/gutenberg/extensions/tiled-gallery/editor.js]: Editor itself already has `wp-polyfill` loaded so this is kinda not needed — should we leave it out and add `wp-polyfill` to dependencies for `editor.js`?

**Comment by blowery** [comment]
We should probably [consider using](https://babeljs.io/docs/en/babel-preset-env#usebuiltins-usage-experimental) `useBuiltIns: usage` for `@babel/preset-env` here instead of `entry`. You don't really need the full polyfill set for the tiled gallery, probably just a couple things.

File: client/gutenberg/extensions/tiled-gallery/editor.js (hunk 1 above)
Comment (sirreal) [reply client/gutenberg/extensions/tiled-gallery/editor.js]: Yes, I'd say so.

File: client/gutenberg/extensions/tiled-gallery/view.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import '@babel/polyfill';
```
Comment (sirreal) [inline client/gutenberg/extensions/tiled-gallery/view.js]: I'd like to compare the size change for the view script with this addition.

If we depend on `wp-polyfill` we can share the cost of the dependency with any other code that may be using it.

I'd like to consider that before moving ahead with this.

**Comment by simison** [comment]
FYI @jeffersonrabb and @ockham since Slideshow block also includes polyfill:

https://github.com/Automattic/wp-calypso/blob/41c7981457c15fdcb20c083a2061204a3961dec4/client/gutenberg/extensions/slideshow/create-swiper.js#L5

We've meant to look into this — ideally SDK builder would just take care of this and polyfill things that are only needed. Calypso would also benefit instead of [adding it manually](https://github.com/Automattic/wp-calypso/blob/cb0ba194b2d7ac6ab32049a33c1a2ae256e20e26/client/boot/polyfills.js)?

**Comment by sirreal** [comment]
> ideally SDK builder would just take care of this

Yes! This is handled for us in Calypso and I'd expect the build tools to provide the same functionality.

File: client/gutenberg/extensions/tiled-gallery/view.js (hunk 2 above)
Comment (simison) [reply client/gutenberg/extensions/tiled-gallery/view.js]: wp-polyfill (v7.0.0) 30.7 KB minified

view.js when building minified production versions:
- without polyfill 7.96 KiB
- with polyfill 87.6 KiB

So looks like we're better enqueing wp-polyfill as a emmediate fix and then later on we'll fix the builder.

Comment (simison) [reply client/gutenberg/extensions/tiled-gallery/view.js]: >     So looks like we're better enqueing wp-polyfill as a immediate fix

https://github.com/Automattic/jetpack/pull/11487

**Comment by simison** [comment]
Closing this as we're going ahead with https://github.com/Automattic/jetpack/pull/11487 instead.

Build tooling polyfill follow-ups: https://github.com/Automattic/wp-calypso/issues/31231

**Pull request closed without merging**